pkg/specgen/specgen.go

```diff
@@ -615,6 +615,8 @@ type ContainerHealthCheckConfig struct {
 	// ("0" value means an infinite log length).
 	// TODO (6.0): In next major release convert it to pointer and use omitempty
 	HealthMaxLogSize uint `json:"healthMaxLogSize"`
+	// HealthImageOverride is a map of healthcheck flags that overrides healthcheck configuration from image.
+	HealthImageOverride map[string]string `json:"HealthImageOverride,omitempty"`
```

Comment:
I am not sure this is the right way to go about. specgen is an external API commitment and we should offer a "sane" json payload for external users. It seems totally confusing to me that there are multiple healtchchecks settings and you need to use one or the other, etc...
cc @mheon 


I don't get why you couldn't just use the existing data to pass it to the server.

Reply:
Tend to concur - Specgen is part of the public container creation API, so we should avoid things that cause confusion where possible. We already have all Healthcheck info in Specgen, so using that instead seems preferable.

Reply:
> I could create a dummy Healthcheck config, which is used to overwrite the values from the dummy Healthcheck to avoid changes in the spacgen.

I think that should be how it works as this seems more logical to users. We basically want overwrite each option one by one and not as a whole on the client.